src/tests/roc_audio/test_depacketizer.cpp

```diff
@@ -300,74 +301,136 @@ TEST(depacketizer, overlapping_packets) {
+
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(packets); n++) {
+        queue.write(packets[n]);
+    }
+
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(packets); n++) {
+        Frame frame = new_frame(SamplesPerPacket);
+        dp.read(frame);
 
-    CHECK(frame.flags() & Frame::FlagSkip);
+        CHECK(frame.flags() & frame_flags[n]);
+    }
 }
 
-TEST(depacketizer, frame_flags_dropped_packets_middle) {
+TEST(depacketizer, frame_flags_packet_drops_full) {
     packet::ConcurrentQueue queue(0, false);
     Depacketizer dp(queue, pcm_decoder, ChMask, false);
 
-    const packet::timestamp_t ts1 = SamplesPerPacket * 3;
-    const packet::timestamp_t ts2 = SamplesPerPacket * 2;
-    const packet::timestamp_t ts3 = SamplesPerPacket * 4;
-
-    queue.write(new_packet(ts1, 0.11f));
-    queue.write(new_packet(ts2, 0.22f));
-    queue.write(new_packet(ts3, 0.33f));
+    queue.write(new_packet(2 * SamplesPerPacket, 0.11f));
+    queue.write(new_packet(1 * SamplesPerPacket, 0.33f));
+    queue.write(new_packet(3 * SamplesPerPacket, 0.33f));
 
-    Frame frame = new_frame(SamplesPerPacket * 3);
+    expect_output(dp, SamplesPerPacket, 0.11f);
+    Frame frame = new_frame(SamplesPerPacket);
     dp.read(frame);
 
-    CHECK(frame.flags() & Frame::FlagSkip);
+    CHECK(frame.flags() & Frame::FlagPacketDrops);
+    CHECK(frame.flags() & Frame::FlagFull);
 }
 
-TEST(depacketizer, frame_flags_no_dropped_packets) {
+TEST(depacketizer, frame_flags_empty_full) {
     packet::ConcurrentQueue queue(0, false);
     Depacketizer dp(queue, pcm_decoder, ChMask, false);
 
-    const packet::timestamp_t ts1 = SamplesPerPacket;
-    const packet::timestamp_t ts2 = SamplesPerPacket * 2;
+    packet::PacketPtr packets[] = {
+        new_packet(SamplesPerPacket, 0.11f),
+        new_packet(SamplesPerPacket * 2, 0.11f),
+        NULL,
+        new_packet(SamplesPerPacket * 5, 0.11f),
+        new_packet(SamplesPerPacket * 7, 0.11f),
+    };
+
+    size_t frame_sizes[] = {
+        SamplesPerPacket,
+        SamplesPerPacket * 2,
+        SamplesPerPacket,
+        SamplesPerPacket * 2,
+        SamplesPerPacket,
+    };
 
-    queue.write(new_packet(ts1, 0.11f));
-    queue.write(new_packet(ts2, 0.22f));
+    unsigned frame_flags[] = {
+        Frame::FlagFull,
+        0,
+        Frame::FlagEmpty,
+        0,
+        Frame::FlagFull,
+    };
 
-    Frame frame = new_frame(SamplesPerPacket * 2);
-    dp.read(frame);
+    CHECK(ROC_ARRAY_SIZE(packets) == ROC_ARRAY_SIZE(frame_sizes));
+    CHECK(ROC_ARRAY_SIZE(frame_sizes) == ROC_ARRAY_SIZE(frame_flags));
+
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(packets); n++) {
+        if (packets[n] != NULL) {
+            queue.write(packets[n]);
+        }
 
-    CHECK(!(frame.flags() & Frame::FlagSkip));
+        Frame frame = new_frame(frame_sizes[n]);
+        dp.read(frame);
+
+        if (frame_flags[n] == Frame::FlagFull) {
+            CHECK(frame.flags() & Frame::FlagFull);
```

Comment:
Same note here, why not just listing and checking the exact flags?

Reply:
Yep, I've fixed it.